axelor-production/src/main/resources/views/PurchaseOrder.xml

```diff
@@ -43,4 +49,75 @@
     <domain>:id member of self.purchaseOrderSet</domain>
   </action-view>
 
+  <action-group name="action-group-purchase-order-partner-onchange"
+    id="action-group-production-purchase-order-partner-onchange">
+    <action name="action-purchase-order-record-partner"/>
+    <action name="action-purchase-order-method-get-stock-location"
+      if="__config__.app.isApp('supplychain') &amp;&amp; (tradingName?.receiptDefaultStockLocation == null || stockLocation == null)"/>
+    <action name="action-purchase-order-method-get-from-stock-location"
+      if="__config__.app.isApp('supplychain')"/>
+    <action name="action-purchase-order-attrs-hide-contact-partner"/>
+    <action name="action-purchase-order-method-fill-company-bank-details"/>
+    <action name="action-purchase-order-method-fill-price-list"/>
+    <action name="action-purchase-order-attrs-interco"/>
+    <action name="action-purchase-order-record-interco"/>
+    <action name="action-purchase-order-record-set-group-product-on-printing"/>
+    <action name="action-purchase-order-method-empty-fiscal-position-if-not-compatible"/>
+    <action name="action-purchase-order-method-fiscal-position-on-change"/>
+    <action name="action-purchase-order-method-compute"/>
+    <action name="action-purchase-order-record-set-type-select"/>
+  </action-group>
```

Comment:
the action group is production should call the the action group in purchase, else we are duplicating which actions must be called on partner change, which is difficult to maintain.

Reply:
Removed duplication and override of this action, instead I directly override `action-purchase-order-record-set-type-select` which is what I needed (Don't know why I didn't do it from starter)